DuggaSys/tests/showDuggaservice_test.php

```diff
@@ -0,0 +1,201 @@
+            'haspwd' => 'asdfasdf',
+            'answer' => 'im updated',
+            'moment' => '<!query-before-test-3!><*[0][listentries]*>'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'submit dugga' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
+        'query-before-test-1' => "INSERT INTO course(cid, creator) VALUES (9999, 1);",
+        'query-before-test-2' => "INSERT INTO listentries(cid,entryname,link,kind,pos,creator,visible,vers,gradesystem,highscoremode,feedbackenabled,feedbackquestion) VALUES (9999, 'Inserttobedeleted', 'UNK', 4, 12, 2, 1, 1337, 1, 1, 0, 'UNK');",
+        'query-before-test-3' => "SELECT MAX(lid) FROM listentries",
+        'query-after-test-1' => "DELETE FROM userAnswer ORDER BY aid DESC LIMIT 1;",
+        'query-after-test-2' => "DELETE FROM listentries WHERE cid = 9999;",
+        'query-after-test-3' => "DELETE FROM course WHERE cid = 9999;",
+        'query-variables' => "moment",
+        'service' => 'https://cms.webug.se/root/G2/students/"a21oscgu"/LenaSYS/DuggaSys/showDuggaservice.php',
+        'service-data' => serialize(array( // Data that service needs to execute function
+            'opt' => 'SAVDU',
+            'cid' => '9999',
+            'coursevers' => '52432',
+            'duggaid' => '1',
+            'moment' => '<!query-before-test3!> <*[0][listentries]*>',
+            'variant' => '3',
+            'hash' => 'ghj1ghj2',
+            'haspwd' => 'asddasdd',
+            'answer' => 'NULL'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'super-view data from useranswer on hash' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
+        'service' => 'https://cms.webug.se/root/G2/students/"a21oscgu"/LenaSYS/DuggaSys/showDuggaservice.php',
```

Comment:
missing pre-querys. course, listentries and useranswer. Follow the test-case file to get these, since most of the others you have done so far are correct.